pkg/platform/baremetal/template.go

```diff
@@ -30,6 +30,13 @@ module "bare-metal-{{.ClusterName}}" {
   # Enable TLS Bootstrap
   enable_tls_bootstrap = {{ .EnableTLSBootstrap }}
 
+  {{- if .EncryptPodTraffic }}
+  encrypt_pod_traffic = {{ .EncryptPodTraffic }}
+  network_mtu = {{ .NetworkMTU }} - 40 # Wireguard overhead
```

Comment:
I don't think this makes sense. IIUC we have two possible values for the `network_mtu` knob:

1. Physical network MTU minus 20 (when IP-IP is used)
2. Physical network MTU minus 40 (when WireGuard is used)

We currently expose a knob called `network_mtu` which has a default value of `1480`. I'm assuming that right now this knob means "the actual MTU to use" as opposed to "the MTU of the **physical** network".

I think we want to do one of the following:

- Decide that `network_mtu` means "the actual MTU to use" and then when WireGuard is used we explicitly set it to `1440`.
- Decide that `network_mtu` means "the MTU of the physical network", set the default to `1500` (for Ethernet) and internally calculate the final value as `network_mtu - 20` for IP-IP and `network_mtu - 60` for WireGuard.

Subtracting 40 from the **IP-IP** value doesn't make sense IMO.

Reply:
@johananl is right, except that Wireguard use 60 bytes header: https://docs.projectcalico.org/networking/mtu#determine-mtu-size. It would be good to clarify the meaning of this flag. I think we could refine it as physical network MTU and then substract 20 from it by default, which preserves existing behavior.

Reply:
Updated